bin/cloud_build.template.json

```diff
@@ -0,0 +1,63 @@
+{
```

Comment:
Could you use yaml. That's what we use everywhere.

Reply:
The REST API only seems to accept json.  I anticipated this preference and I did try to make this file into yaml but the REST API failed complaining it wasn't valid json.  It's possible that I didn't try hard enough to report the type encoding of the POST content but the docs don't mention anything besides json.

Reply:
I tried a bit harder by specifying:
-H "Content-Type: application/yaml"
(I also tried "application/x-yaml" and "text/yaml") but the build request still immediately fails so I'm guessing the website and file parsing frontends do a conversion of yaml to json.  It's not that I was heavily set on json (in fact the appending of content at the end of the template would have been easier and cleaner if I didn't have to handle the trailing } characters). 

{
  "error": {
    "code": 400,
    "message": "Invalid JSON payload received. Unexpected token.\nsteps:\n- name: gcr.i\n^",
    "status": "INVALID_ARGUMENT"
  }
}

Reply:
Got it

Reply:
Is bin the right directory ? Maybe tools/ ? There is a bit of pollution problem...

Reply:
I haven't seen enough consistency in istio to find a pattern to follow.  Prior to mono-repo each component's bin/ directory had a yaml file that was analogous to this file.  I followed this model with this json file (though it's hard to tell because the yaml files were already deleted as part of the repo unification).  There's also the release/ directory, and I moved create_release_archives.sh from there to bin/ since there's now only 1 file left in release/ (downloadIstioCandidate.sh) and I doubt it's worth having a separate release/ anymore. I could just place in all in release instead, but the scripts can be used for other purposes too (possibly leveraged by CI or other frameworks).  The main intent for cloud build is to do "official" daily releases but a dev can also use it to make a test build of their own changes (the main effort is the 1-time setup of a Google Cloud project and a manifest repo that points to the dev's forked repo).  Consequently I was hesitant to place the files in a less-generic location, particularly since it could also make it easier for people to overlook the files and break things in future changes.

Reply:
I think release/ makes sense
(it has to do with OWNERS too)